Doc/whatsnew/3.13.rst

```diff
@@ -125,6 +124,11 @@ Other Language Changes
   equivalent of the :option:`-X frozen_modules <-X>` command-line option.
   (Contributed by Yilei Yang in :gh:`111374`.)
 
+* The :func:`ssl.create_default_context` API now includes
+  :data:`ssl.VERIFY_X509_PARTIAL_CHAIN` and :data:`ssl.VERIFY_X509_STRICT`
+  in its default flags.
```

Comment:
Can you please explain (give instructions) how to get the previous behavior?

Also, how can someone write future-proof code by switching to these defaults in Python 3.12 and older? Maybe also provide instructions for that?

Reply:
Would you mind to explain which kinds of certificates are no longer accepted? If someone is impacted, how certificates should be generated to fit with stricter security policy?

Reply:
> Can you please explain (give instructions) how to get the previous behavior?

Sure, I can add a note callout similar to the one for SSLv3.

> Also, how can someone write future-proof code by switching to these defaults in Python 3.12 and older? Maybe also provide instructions for that?

The simplest way would probably be to either use `create_default_context()` or `SSLContext()` and twiddle `ctx.verify_flags` directly. 

Do you think it makes sense to include that in the `ssl.rst` docs? It looks like there isn't precedent for doing so (unlike the callout for SSLv3). If so, I'm happy to do it 🙂 

> Would you mind to explain which kinds of certificates are no longer accepted? If someone is impacted, how certificates should be generated to fit with stricter security policy?

This has two parts:

* `VERIFY_X509_PARTIAL_CHAIN` makes chain building strictly _more_ permissive, not less. Specifically, it tells OpenSSL to build chains the way RFC 5280 tells it to, not the bespoke way it does by default. So this flag _alone_ shouldn't cause any certificate rejections, only discoveries of more valid chains 🙂 -- no changes should be required on users' parts.
* `VERIFY_X509_STRICT` tells OpenSSL to validate X.509 certificates per the RFC 5280 profile, which has more extension/criticality requirements than OpenSSL checks by default. In practice, this should have no effect on >99% of users: most root programs use 5280 as their baseline, and the Web PKI's CABF rules are also built on 5280.

TL;DR: No generation changes need to happen for `VERIFY_X509_PARTIAL_CHAIN`. For `VERIFY_X509_STRICT`, the small number of users doing things like `openssl req` manually should ensure that their generated certs conform to RFC 5280 (which they already need to, if they're being used in browsers or other contexts where 5280 or a superset is assumed).

Reply:
> Do you think it makes sense to include that in the `ssl.rst` docs?

Doing it in the what's new makes sense for sure - It doesn't hurt to also include it in the `ssl.rst` versionchanged section. People come at docs from multiple angles depending on what they're trying to do or when and how they are debugging things.

Reply:
Either link to ssl doc which explains how to get Python 3.12 behavior, or repeat the instructions here. Something like:

```py
context.verify_flags &= ~(_ssl.VERIFY_X509_PARTIAL_CHAIN | _ssl.VERIFY_X509_STRICT)
```

Reply:
Added a callout similar to the one for SSLv3 here: 4ae44d1e7b

That'll get linked via the ref to `:func:ssl.create_default_context` -- do you think another link makes sense, or is that sufficient?

Reply:
Would it be possible to add a link from here to the this new doc? Or copy/paste it here?